### PR TITLE
Update system during deployment

### PR DESCRIPTION
**Pull request by varkoly**

### CODE REVIEW

**Comment by arbulu89** [comment]
@varkoly And other important thing. The update cannot be made if `qa_mode` is set. @ldevulder , @juadk Could you confirm this?

**Comment by varkoly** [comment]
@arbulu89  That is why I've pushed the update first into the registration.sls. Update makes only sence if registration was done. Now I'll build in the same conditions like there.

**Comment by MalloZup** [comment]
when merging this, use the squash commits option :pray:

File: salt/default/pkgs.sls [hunk 1]
```diff
@@ -7,4 +7,5 @@ install_additional_packages:
 {% endfor %}
     - require:
       - sls: default.repos
+      - sls: default.update
```
Comment (arbulu89) [inline salt/default/pkgs.sls]: I wouldn't personally add this entry. Actually, I would get rid of tis `pkgs` state. Nobody uses it.

**Comment by varkoly** [comment]
> @varkoly @diegoakechi It looks good to me. As far as I see, it shouldn't bring any issue. Have you tested? At least in libvirt?

I'm just doing it. I'll report if the test has worked.

**Comment by diegoakechi** [comment]
@varkoly This PR should be done to the `develop` branch and not directly to `master`.

**Comment by varkoly** [comment]
I'll resubmit it into devel branch after I've tested IT.

**Pull request closed without merging**